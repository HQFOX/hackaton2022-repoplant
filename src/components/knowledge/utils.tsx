import {
  AssetInventoryMetadata,
  AssetInventoryConfiguration,
} from "@hitachivantara/uikit-react-core";

const metadata: AssetInventoryMetadata[] = [
  {
    id: "id1",
    accessor: "title",
    cellType: "alpha-numeric",
  },
  {
    id: "id2",
    accessor: "project",
    cellType: "alpha-numeric",
  },
];

const viewConfiguration: AssetInventoryConfiguration = {
  breakpoints: { xs: 12, sm: 6, md: 4, lg: 3, xl: 3 },
  columnConfiguration: [
    {
      style: { paddingLeft: "10px", textAlign: "start" },
    },
    {
      title: "Title",
      style: { textAlign: "start", minWidth: "120px" },
    },
    {
      title: "Project",
      style: { textAlign: "start", minWidth: "80px" },
    },
    {
      title: "Information Type",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Uploaded By",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Date",
      style: {
        textAlign: "start",
      },
    },
  ],
};

const configuration = {
  metadata,
  viewConfiguration,
};

const actions = [{ id: "view", label: "View", disabled: false }];

export { configuration, actions };
