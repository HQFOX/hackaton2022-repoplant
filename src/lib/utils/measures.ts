export const avg = (values: number[]): number => {
  if (!values || values.length === 0) {
    return 0;
  }
  return (
    values.reduce((a, b) => {
      return a + b;
    }) / values.length
  );
};
