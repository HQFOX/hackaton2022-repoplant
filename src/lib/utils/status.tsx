import React, { lazy } from "react";
import { Level5 } from "@hitachivantara/uikit-react-icons";
import { Semantic } from "@hitachivantara/uikit-react-lab";

export const getStatus = (status: any) => {
  const Level = `Level${status}`;
  const statusObj: {
    Icon: React.ReactNode;
    sema: Semantic;
  } = { Icon: <Level5 />, sema: "sema5" };
  return statusObj;
};
