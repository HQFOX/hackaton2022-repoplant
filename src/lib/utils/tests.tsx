import React from "react";
import { MemoryRouter } from "react-router-dom";
import { HvProvider } from "@hv/uikit-react-core";
import ContextProvider from "lib/providers/ContextProvider";

export const withProvider = <P extends {}>(
  Component: React.ComponentType<P>
): React.FC<P> => (props) => (
  <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
    <ContextProvider>
      <HvProvider>
        <Component {...props} />
      </HvProvider>
    </ContextProvider>
  </MemoryRouter>
);
