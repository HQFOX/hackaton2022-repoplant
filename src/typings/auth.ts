export interface AuthCredentials {
  username: string;
  password: string;
}

export interface AuthContextType {
  isAuthed: boolean;
  authStatus: string;
  login: (value: AuthCredentials) => void;
  logout: () => void;
}

export interface RecoverContextType {
  recoverStatus: string;
  activeForm: string;
  setActiveForm: (value: string) => void;
  recover: (value: string) => void;
}
