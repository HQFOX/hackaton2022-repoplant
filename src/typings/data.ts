// Use enums for better autocompletion of action type names.
//
// We use the `@@context/ACTION_TYPE` convention,
// to follow the convention of Redux's `@@INIT` action.
export enum DataActions {
  SET_ASSETS = "@@data/SET_ASSETS",
  SET_TABLE = "@@data/SET_TABLE",
  SET_LINE = "@@data/SET_LINE",
  SET_BAR = "@@data/SET_BAR",
}

// Declare state types with `readonly` modifier to get compile time immutability.
// https://github.com/piotrwitek/react-redux-typescript-guide#state-with-type-level-immutability
export type AssetsData = object[];
export type TableData = object[];
export type LineData = object[];
export type BarData = object[];

export interface DataState {
  assets: AssetsData;
  table: TableData;
  line: LineData;
  bar: BarData;
}
