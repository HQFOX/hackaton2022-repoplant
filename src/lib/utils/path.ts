import pages from "lib/api/pages";
import { Page } from "typings/pages";

const getActivePath = (
  pathname: string,
  data: Page[] = pages
): Page | undefined => {
  let activePath;

  data.forEach((item) => {
    const hasPath = pathname.includes(item.path);

    if (hasPath) {
      activePath = item.data ? getActivePath(pathname, item.data) : { ...item };
    }
  });

  return activePath;
};

const isTopLevelPage = (pathname: string): boolean => {
  return !!pages.find((item) => item.path === pathname);
};

export { getActivePath, isTopLevelPage };
