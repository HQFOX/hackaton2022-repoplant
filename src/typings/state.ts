import { AuthState } from "./auth";

export type DeepPartial<T> = T extends Function
  ? T
  : T extends object
  ? { [P in keyof T]?: DeepPartial<T[P]> }
  : T;

// The top-level state object
export interface AppState {
  readonly auth: AuthState;
}

export type TestAppState = DeepPartial<AppState>;
