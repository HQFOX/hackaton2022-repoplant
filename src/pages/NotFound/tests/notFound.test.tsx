import React from "react";
import { render } from '@testing-library/react';
import { withStoreProvider } from "lib/utils/tests";
import { TestAppState } from "typings/state";
import NotFound from "..";

describe("<NotFound />", () => {
  let snapshot;

  beforeEach(() => {
    const WithStoreProvider = withStoreProvider(NotFound, {
      router: { location: { pathname: "/" } },
      auth: { isAuthed: false },
      pages: { data: [] },
      logout: jest.fn()
    } as TestAppState);
    snapshot = render(<WithStoreProvider />);
  });

  test("it matches the snapshot", () => {
    expect(snapshot.container).toMatchSnapshot();
  });
});
