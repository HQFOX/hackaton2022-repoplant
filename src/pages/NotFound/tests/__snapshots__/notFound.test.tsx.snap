// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> it matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-section-2"
  >
    <div
      class="MuiContainer-root-9 HvContainer-root makeStyles-container-3 MuiContainer-maxWidthXl-16 HvContainer-maxWidthXl"
    >
      <div
        class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-spacing-xs-2-41"
      >
        <div
          class="MuiGrid-root-17 MuiGrid-item-19 MuiGrid-grid-xl-12-119"
        >
          <h1
            class="NotFound-title-1 HvTypography-root HvTypography-3xlTitle"
          >
            pages.notFound.title
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
